pkg/provider/nomad/nomad.go

```diff
@@ -183,26 +190,51 @@ func (p *Provider) Provide(configurationChan chan<- dynamic.Message, pool *safe.
 			ctx, cancel := context.WithCancel(ctxLog)
 			defer cancel()
 
-			// load initial configuration
-			if err := p.loadConfiguration(ctx, configurationChan); err != nil {
-				return fmt.Errorf("failed to load initial nomad services: %w", err)
+			serviceEventsChan, err := p.pollOrWatch(ctx)
```

Comment:
In the polling, if I am not mistaken we will not load the initial configuration until the first tick, which was not the case previously.